opencv_ai/splice/io.h

```diff
@@ -0,0 +1,166 @@
+#ifndef OPENCV_AI_SPLICE_IO_H_
+#define OPENCV_AI_SPLICE_IO_H_
+
+#include <opencv_ai/splice/component.h>
+#include <boost/signals2.hpp>
+
+namespace opencv_ai {
+
+template <typename MessageType>
+class ChannelService : public boost::asio::io_service::service {
+ public:
+  ChannelService(boost::asio::io_service& io_service)
+      : boost::asio::io_service::service(io_service) {}
+
+  /// The unique service identifier.
+  static boost::asio::io_service::id id;
+  // Required by base class.
+  void shutdown_service() override {}
+
+  typedef boost::signals2::signal<void(MessageType)> Signal;
+
+  Signal& LookUpSignal(const std::string& channel_name) {
+    return signals_[channel_name];
+  }
+
+ private:
+  std::map<std::string, Signal> signals_;
+};
+
+template <typename MessageType>
+boost::asio::io_service::id ChannelService<MessageType>::id;
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
```

Comment:
Even though these are in a namespace, we should probably move these using statements into a function scope, or fully qualify them as needed.